### PR TITLE
Write batch record after writing individual component records

### PR DESCRIPTION
**Pull request by samfrench**
When having 30+ models in one folder, the broker looks up a component before it has been written. This is caused by the batch record being written first.

This PR moves the batch record writing to after each component record has been written. If the record is successful then an `AWS::Core::Response` is returned, and we can call `successful?` on it.

https://jira.dev.bbc.co.uk/browse/RESFRAME-1018

### CODE REVIEW

**Comment by samfrench** [comment]
@Tarqwyn and @zygotic99, here is a fix for the batch writing, to be done last. This checks if the individual records are successful before writing the batch record.

It would be good to get your thoughts on this. This would be the better way of achieving the fix, over PAL amendments, but it's up to you if we merge after or not.

cc @ettomatic

File: lib/alephant/publisher/queue/writer.rb [hunk 1]
```diff
@@ -38,15 +38,22 @@ def parser
         end
 
         def run!
-          seq_for(config[:renderer_id]).validate(message, &process_components)
+          if component_records_write_successful?
+            seq_for(config[:renderer_id]).validate(message) do
+              # block needed in sequencer. Need to make optional with `block.given?`
+              # https://github.com/BBC-News/alephant-sequencer/blob/master/lib/alephant/sequencer/sequencer.rb#L41
+            end
```
Comment (Tarqwyn) [inline lib/alephant/publisher/queue/writer.rb]: Am I right in thinking we need to make this change to sequencer or that it is merely recommended it.

**Comment by Tarqwyn** [comment]
We could test this via a rehearsal and spin back the version on the elections renderers if it doesn't work..

File: lib/alephant/publisher/queue/writer.rb (hunk 1 above)
Comment (samfrench) [inline lib/alephant/publisher/queue/writer.rb]: It is a recommendation for when we are next in Sequencer. It's more of a `todo` but doesn't have to be done now.

**Pull request merged**